### ISSUE TITLE
`reference` not `inline` result types.

### ISSUE THREAD

**Issue by oscartbeaumont** [oscartbeaumont/tauri-specta#72]
https://discord.com/channels/616186924390023171/731495028677148753/1201755428728606780

**Comment by nobane** [comment]
The link points you to my message about:

```rs
#[derive(Serialize, Type)]
struct InitData {
  files: Vec<FileDetail>,
}
#[tauri::command]
#[specta::specta]
pub async fn get_init_data() -> Result<InitData, ()> { 
//...
}
```
makes
```ts
async getInitData() : Promise<__Result__<{  files: FileDetail[]; }> {
```
but it would be nice if it was actually 
```ts
async getInitData() : Promise<__Result__<InitData> {
```

**Comment by oscartbeaumont** [comment]
This was probably fixed by https://github.com/oscartbeaumont/specta/issues/191

**Comment by Sharpiro** [comment]
> This was probably fixed by [oscartbeaumont/specta#191](https://github.com/oscartbeaumont/specta/issues/191).
> 
> Hopefully a new Specta release and upgrading Tauri Specta will be enough to fix it.

Can we get a new Tauri Specta RC build to see if this works now?

**Comment by oscartbeaumont** [comment]
I can confirm it is fixed from my testing so i'm going to close this issue.

If you checkout the [docs here](https://github.com/oscartbeaumont/tauri-specta/blob/main/docs/v2.md) they will help you get setup with a fixed version. I will do a release once the bugs in #86 are sorted out.
